pkg/admission/validate_bucketclass.go

```diff
@@ -0,0 +1,207 @@
+		arRequest: &arRequest,
+		Logger:    logrus.WithField("admission bucketclass validation", arRequest.Request.Namespace),
+	}
+	return bcv
+}
+
+// ValidateBucketClass call appropriate validations based on the operation
+func (bcv *BucketClassValidator) ValidateBucketClass() admissionv1.AdmissionReview {
+	bcv.arResponse = &admissionv1.AdmissionReview{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AdmissionReview",
+			APIVersion: "admission.k8s.io/v1",
+		},
+		Response: &admissionv1.AdmissionResponse{
+			UID:     bcv.arRequest.Request.UID,
+			Allowed: true,
+			Result: &metav1.Status{
+				Message: "allowed",
+			},
+		},
+	}
+
+	switch bcv.arRequest.Request.Operation {
+	case admissionv1.Create:
+		bcv.ValidateCreate()
+	case admissionv1.Delete:
+		bcv.ValidateDelete()
+	default:
+		bcv.Logger.Error("Failed to identify bucketclass operation type")
+	}
+	return *bcv.arResponse
+}
+
+// SetValidationResult responsible of assinging the return values of a validation into the response appropriate fields
+func (bcv *BucketClassValidator) SetValidationResult(isAllowed bool, message string) {
+	bcv.arResponse.Response.Allowed = isAllowed
+	bcv.arResponse.Response.Result.Message = message
+}
+
+// DeserializeBC extract the bucketclass object from the request
+func (bcv *BucketClassValidator) DeserializeBC(rawBS []byte) *nbv1.BucketClass {
+	BC := nbv1.BucketClass{}
+	if err := json.Unmarshal(rawBS, &BC); err != nil {
+		bcv.Logger.Error("error deserializing bucketclass")
+	}
+	return &BC
+}
+
+// ValidateCreate runs all the validations tests for CREATE operations
+func (bcv *BucketClassValidator) ValidateCreate() {
+	bcv.BucketClass = bcv.DeserializeBC(bcv.arRequest.Request.Object.Raw)
+
+	if ok, message := bcv.ValidateQuota(); !ok {
+		bcv.SetValidationResult(ok, message)
+		return
+	}
+	if bcv.BucketClass.Spec.NamespacePolicy != nil {
+		if ok, message := bcv.ValidateNSFSSingleBC(); !ok {
+			bcv.SetValidationResult(ok, message)
+			return
+		}
+	} else {
+		if ok, message := bcv.ValidateTiersNumber(); !ok {
+			bcv.SetValidationResult(ok, message)
+			return
+		}
+	}
+}
+
+// ValidateDelete runs all the validations tests for DELETE operations
+func (bcv *BucketClassValidator) ValidateDelete() {
+	bcv.BucketClass = bcv.DeserializeBC(bcv.arRequest.Request.OldObject.Raw)
+	if ok, message := bcv.ValidateBucketclassDeletion(); !ok {
+		bcv.SetValidationResult(ok, message)
+		return
+	}
+}
+
+// ValidateTiersNumber validates the provided number of tiers is 1 or 2
+func (bcv *BucketClassValidator) ValidateTiersNumber() (bool, string) {
+	numTiers := len(bcv.BucketClass.Spec.PlacementPolicy.Tiers)
+	if numTiers != 1 && numTiers != 2 {
+		return false, "Unsupported number of tiers, bucketClass supports only 1 or 2 tiers"
+	}
+	return true, "allowed"
+}
+
+// ValidateNSFSSingleBC validates that bucketclass configured to NS of type NSFS it will only be of type Single.
+func (bcv *BucketClassValidator) ValidateNSFSSingleBC() (bool, string) {
```

Comment:
don't we want to test for all namespace bucketclasses that their namespacestores exists and ready? not only nsfs and not just single
besides that, nsfs can be currently only of type single but it's another validation